subprojects/docs/src/docs/userguide/embedding.adoc

```diff
@@ -66,10 +66,73 @@ The main entry point to the Tooling API is the api:org.gradle.tooling.GradleConn
 [[sec:embedding_compatibility]]
 === Gradle version and Java version compatibility
 
-The current version of the Tooling API supports running builds using Gradle versions 1.2 and later.
+==== Provider side
+
+The current version of the Tooling API supports running builds using Gradle versions 2.6 and later. Gradle version between 1.2 and 2.6 is still supported but will be removed in Gradle 5.0, which you should avoid using.
+
+.Tooling API provider support
+
+|===
+| Tooling API provider (Gradle runtime) | Current Tooling API | Tooling API 5.0
+
+| version < 1.2
+| Not supported
+| Not supported
+
+| 1.2 <= version < 2.6
+| Deprecated
+| Not supported
+
+| version > 2.6
+| Supported
+| Supported
+|===
 
 You should note that not all features of the Tooling API are available for all versions of Gradle. For example, build cancellation is only available when a build uses Gradle 2.1 and later. Refer to the documentation for each class and method for more details.
 
-The current Gradle version can be used from Tooling API versions 2.0 or later.
+=== Client side
+
+The current Gradle version can be used from Tooling API versions 3.0 or later. Tooling API version between 2.0 and 3.0 is still supported but will be removed in Gradle 5.0, which you should avoid using.
```

Comment:
❌ This sentence is confusing because of a grammatical issue: "which you should avoid using" comes after "Gradle 5.0", which means that I shouldn't use Gradle 5.0. That's not the message we want.

I'd just remove that last part of the sentence. It is obvious that users shouldn't use deprecated things.